packages/catalogue/src/catalogue.ts

```diff
@@ -0,0 +1,29 @@
+import type { CredentialsProvider } from '@tidal-music/common';
+import createClient from 'openapi-fetch';
+
+import type { paths } from './catalogueAPI';
+
+let authToken: string | undefined = undefined;
+
+/**
+ * Initializes the client with the provided credentials.
+ *
+ * @param credentialsProvider The credentials provider.
+ */
+export async function initCatalogueClient(
+  credentialsProvider: CredentialsProvider,
+) {
+  const credentials = await credentialsProvider.getCredentials();
+  authToken = credentials.token;
+  //TODO: listen for changes to the token
```

Comment:
If we do a new release of auth you could try the new bus that sends the credentials alongside `credentialsUpdated`

Reply:
(not needed as now it is async)